### PR TITLE
refactor: implement databricks volumes v2 dest connector

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

File: test_unstructured_ingest/python/test-databricks-volumes.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    volume: str,
+    volume_path: str,
+):
+    client = WorkspaceClient(host=host, username=username, password=password)
+
+    for file in client.files.list_directory_contents(
+        _get_volume_path(catalog, volume, volume_path)
+    ):
+        client.files.delete(file.path)
+    client.files.delete_directory(_get_volume_path(catalog, volume, volume_path))
```
Comment (potter-potter) [inline test_unstructured_ingest/python/test-databricks-volumes.py:75]: This is awesome! Nice.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -412,6 +412,10 @@ jobs:
         ASTRA_DB_TOKEN: ${{secrets.ASTRA_DB_TOKEN}}
         ASTRA_DB_ENDPOINT: ${{secrets.ASTRA_DB_ENDPOINT}}
         CLARIFAI_API_KEY: ${{secrets.CLARIFAI_API_KEY}}
+        DATABRICKS_HOST: ${{secrets.DATABRICKS_HOST}}
```
Comment (potter-potter) [inline .github/workflows/ci.yml:415]: We'll probably leave these out of the CI.yml

File: test_unstructured_ingest/python/test-databricks-volumes.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        client.files.list_directory_contents(_get_volume_path(catalog, volume, volume_path))
+    )
+
+    assert len(files) == 1
+
+    resp = client.files.download(files[0].path)
+    data = json.loads(resp.contents.read())
+
+    # Are these assertions too opinionated?
+    # We don't need tests failing when we change how we process the data
```
Comment (potter-potter) [inline test_unstructured_ingest/python/test-databricks-volumes.py:43]: If we are just running this every so often, this is great. 

Let's come to consensus on if we want to run databricks-volumes.sh on CI. If so, then maybe.. but I think it's better to be more opinionated.

Think of the standard CI expected-structured-output. They are super detailed/opinionated.

File: unstructured/ingest/v2/processes/connectors/databricks_volumes.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    host: Optional[str] = None
+
+
+@dataclass
+class DatabricksVolumesUploaderConfig(UploaderConfig):
+    volume: str
+    catalog: str
+    volume_path: Optional[str] = None
+    overwrite: bool = False
+    encoding: str = "utf-8"
```
Comment (potter-potter) [inline unstructured/ingest/v2/processes/connectors/databricks_volumes.py:59]: I don't think this gets used anywhere

Comment (vangheem) [reply unstructured/ingest/v2/processes/connectors/databricks_volumes.py:59]: You're right!

File: test_unstructured_ingest/python/test-databricks-volumes.py (hunk 3 above)
Comment (vangheem) [reply test_unstructured_ingest/python/test-databricks-volumes.py:43]: Should I be more opinionated here and have tests that do exact matches for output?

Comment (potter-potter) [reply test_unstructured_ingest/python/test-databricks-volumes.py:43]: I don't think so. We do super opinionated for source connectors. For destination its usually less opinionated. And then on vector destinations we do a vector search.

What you have is fine. I think for destination we just need to test that its a real file. If there are errors in partitioning,chunking,embedding those should be exposed by source tests anyway.

**Review by potter-potter — APPROVED** [review]
looks good

File: test_unstructured_ingest/python/test-databricks-volumes.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    assert len(data) == 5
+    assert [v["type"] for v in data] == [
+        "UncategorizedText",
+        "Title",
+        "NarrativeText",
+        "UncategorizedText",
+        "Title",
+    ]
```
Comment (potter-potter) [inline test_unstructured_ingest/python/test-databricks-volumes.py:51]: Can you add a quick line to say that it passed so we can see it in the CI logs.

File: test_unstructured_ingest/python/test-databricks-volumes.py (hunk 3 above)
Comment (vangheem) [reply test_unstructured_ingest/python/test-databricks-volumes.py]: Cool! Removed comment :)

File: test_unstructured_ingest/python/test-databricks-volumes.py (hunk 5 above)
Comment (vangheem) [reply test_unstructured_ingest/python/test-databricks-volumes.py:49]: done!

File: unstructured/ingest/v2/processes/connectors/databricks_volumes.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            with open(content.path, "rb") as elements_file:
+                output_path = os.path.join(self.upload_config.path, content.path.name)
+                self.client.files.upload(
+                    file_path=output_path,
+                    contents=elements_file,
+                    overwrite=self.upload_config.overwrite,
+                )
+
+
+add_destination_entry(
```
Comment (potter-potter) [inline unstructured/ingest/v2/processes/connectors/databricks_volumes.py:95]: This has been changed in a recent PR. Roman alerted me. See the other V2 destination connectors.

**Pull request merged**